llvm/lib/Target/AArch64/AArch64InstrInfo.td

```diff
@@ -247,6 +247,11 @@ def HasSMEF16F16orSMEF8F16
 def HasNEONandIsStreamingSafe
     : Predicate<"Subtarget->hasNEON()">,
       AssemblerPredicateWithAll<(any_of FeatureNEON), "neon">;
+// A subset of NEON instructions legal in Streaming SVE mode with +sme2p2.
+// TODO: Change to check for hasSME2p2() once FEAT_SME2p2 is implemented.
+def HasNEONandIsSME2p2StreamingSafe
```

Comment:
nit: At the moment the predicate does not do what the name suggests. Could you use the existing `HasNEON` predicate and instead add a TODO to the places where `HasNEONandIsStreamingSafe` was used incorrectly?